main/src/cgeo/geocaching/models/Waypoint.java

```diff
@@ -479,7 +480,7 @@ private static WaypointType parseWaypointType(final String input, final String l
             }
         }
         for (final WaypointType wpType : WaypointType.ALL_TYPES) {
-            if (lowerInput.contains(wpType.getL10n().toLowerCase(Locale.getDefault()))) {
+            if (lowerInput.contains(wpType.getNameForNewWaypoint().toLowerCase(Locale.getDefault())) || lowerInput.contains(wpType.getL10n().toLowerCase(Locale.getDefault()))) {
```

Comment:
Hm, shouldn't we use different "newWaypointNames" for the two waypoint types then (besides doing two separate loops, if we stay with "stage" as one of them)?

I already had a collision at another place (filling the suggestion view adapter in EditWaypointActivity), but that was easy to mitigate.

Reply:
e. g.: "STAGE" (for physical stages) and "VIRTUAL"?

Reply:
I thought about this topic a lot when I implemented the whole PN parsing issue many month ago.

Imho:
I think it is close to impossible to control this in all translations. Also, interactions between subparts of the different translations would have to be considered. I don't think the "parse PN" use case is important enough to justify such strict manners.

But it should be understandable by the user what is happening, and she should have at least one possibility to specify each waypoint type uniquely in the PN (hence the introduction of the short waypoint type in parenthesis, this will always work uniquely).

In summary, I think we should make a best effort to make this work as expected by the user but not be too strict about it.

Reply:
@eddiemuc 
Do I understand that correctly that you propose to only split up the loop into two, checking the old (longer) names first, and leave the naming as it is in the PR?

Reply:
> @eddiemuc 
> Do I understand that correctly that you propose to only split up the loop into two, checking the old (longer) names first, and leave the naming as it is in the PR?

Yes, that would be my suggestion

Reply:
Ok, thanks, have updated the code accordingly